base/abstractarray.jl

```diff
@@ -2422,7 +2422,7 @@ function hvncat_fill!(A::Array, row_first::Bool, xs::Tuple)
             end
         end
     else
-        for k ∈ eachindex(xs)
+        @inbounds for k ∈ eachindex(xs)
             A[k] = xs[k]
```

Comment:
Would
```suggestion
        for k ∈ eachindex(A, xs)
            A[k] = xs[k]
```
let us avoid `@inbounds` and retain same performance?

Reply:
Wasn't aware of the multi-arg `eachindex`, but looks like no:
```
julia> eachindex([1, 2, 3], (2, 3, 4))
ERROR: MethodError: no method matching keys(::Vector{Int64}, ::Tuple{Int64, Int64, Int64})
```